toolchain/diagnostics/diagnostic_converter.h

```diff
@@ -15,9 +15,16 @@ namespace Carbon {
 template <typename LocationT>
 class DiagnosticConverter {
  public:
+  using ContextFnT = llvm::function_ref<void(
```

Comment:
```suggestion
  // Callback type used to report context messages from the conversion.
  // Note that the first parameter type is DiagnosticLocation rather than
  // LocationT, because it's an output of the conversion.
  using ContextFnT = llvm::function_ref<void(
```

Reply:
Done, with some edits to specify ConvertLocation